manifests/config/apache.pp

```diff
@@ -101,12 +106,88 @@
     $suburi_without_slash = join(values_at($suburi_parts, ["3-${suburi_parts_count}"]), '/')
     if $suburi_without_slash {
       $suburi = "/${suburi_without_slash}"
+    } else {
+      $suburi = undef
+    }
+  } else {
+    $suburi = undef
+  }
+
+  if $passenger {
+    $passenger_options = {
+      'passenger_app_root' => $app_root,
+      'passenger_min_instances' => $passenger_min_instances,
+      'passenger_start_timeout' => $passenger_start_timeout,
+      'passenger_ruby' => $passenger_ruby,
+    }
+    $passenger_http_prestart = $passenger_prestart ? {
+      true  => "http://${servername}:${server_port}",
+      false => undef,
+    }
+    $passenger_https_prestart = $passenger_prestart ? {
+      true  => "https://${servername}:${server_ssl_port}",
+      false => undef,
+    }
+
+    if $suburi {
+      $custom_fragment = template('foreman/_suburi.conf.erb')
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $proxy_http_options = {}
+    $proxy_https_options = {}
+
+    if $app_root {
+      file { ["${app_root}/config.ru", "${app_root}/config/environment.rb"]:
+        owner => $user,
+      }
+    }
+  } else {
+    $passenger_options = {}
+    $passenger_http_prestart = undef
+    $passenger_https_prestart = undef
+
+    if $suburi {
+      $custom_fragment = undef
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $proxy_http_options = {
+      'proxy_preserve_host' => true,
+      'proxy_add_headers'   => true,
+      'request_headers' => [
+        'set X_FORWARDED_PROTO "http"',
+      ],
+      'proxy_pass'          => {
+        'no_proxy_uris' => $proxy_no_proxy_uris,
+        'path'          => pick($suburi, '/'),
+        'url'           => $proxy_backend,
+        'params'        => $proxy_params,
+      },
+    }
+
+    $proxy_https_options = {
+      'ssl_proxyengine' => true,
+      'request_headers' => [
+        'set X_FORWARDED_PROTO "https"',
+        'set SSL_CLIENT_S_DN "%{SSL_CLIENT_S_DN}s"',
+        'set SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"',
+        'set SSL_CLIENT_VERIFY "%{SSL_CLIENT_VERIFY}s"',
+      ],
+    }
+
+    if $selinux or ($facts['selinux'] and $selinux != false) {
```

Comment:
Why this second boolean instead of just relying on `$facts['selinux']`?

Reply:
We have a parameter that allows you to turn off selinux and this copies the pattern in a slightly cleaner way:
https://github.com/theforeman/puppet-foreman/blob/458f93ad21937a7583cad57927e42917cddd8447/manifests/config.pp#L91-L96

We should clean that up as well to a more modern coding standard.

Reply:
oh, ok :)